src/Middleware/ResponseCompression/test/ResponseCompressionMiddlewareTest.cs

```diff
@@ -1300,7 +1321,55 @@ private static void CheckResponseCompressed(HttpResponseMessage response, long?
         Assert.True(containsVaryAcceptEncoding);
         Assert.False(response.Content.Headers.TryGetValues(HeaderNames.ContentMD5, out _));
         Assert.Single(response.Content.Headers.ContentEncoding, expectedEncoding);
-        Assert.Equal(expectedBodyLength, response.Content.Headers.ContentLength);
+
+        // Test functionality instead of exact byte counts
+        CheckCompressionFunctionality(response, expectedEncoding).GetAwaiter().GetResult();
+    }
+
+    private static async Task CheckCompressionFunctionality(HttpResponseMessage response, string expectedEncoding)
+    {
+        var compressedBytes = await response.Content.ReadAsByteArrayAsync();
+
+        // Handle HEAD requests - no body to decompress
+        if (response.RequestMessage?.Method == HttpMethod.Head)
+        {
+            return;
+        }
+
+        // Decompress and verify content matches original
+        string decompressedContent;
+
+        if (expectedEncoding == "gzip")
+        {
+            using var compressedStream = new MemoryStream(compressedBytes);
+            using var gzipStream = new GZipStream(compressedStream, CompressionMode.Decompress);
+            using var reader = new StreamReader(gzipStream);
+            decompressedContent = await reader.ReadToEndAsync();
+        }
+        else if (expectedEncoding == "br")
+        {
+            using var compressedStream = new MemoryStream(compressedBytes);
+            using var brotliStream = new BrotliStream(compressedStream, CompressionMode.Decompress);
+            using var reader = new StreamReader(brotliStream);
+            decompressedContent = await reader.ReadToEndAsync();
+        }
+        else
+        {
+            throw new ArgumentException($"Unsupported encoding: {expectedEncoding}");
+        }
+
+        // Verify decompressed content matches what we expect
+        var expectedContent = new string('a', 100);
```

Comment:
nit: This should be passed into the function. Tests aren't guaranteed to use this as the content.